#!/usr/bin/env python
PACKAGE = "mmuav_control"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("x_kp", double_t, 0, "x ctl PID P gain", 0.5, 0, 100)
gen.add("x_ki", double_t, 0, "x ctl PID I gain", 0.1, 0, 100)
gen.add("x_kd", double_t, 0, "x ctl PID D gain", 0.0, 0, 100)
gen.add("vx_kp", double_t, 0, "x rate ctl PID P gain", 20, 0, 1000)
gen.add("vx_ki", double_t, 0, "x rate ctl PID I gain", 20, 0, 1000)
gen.add("vx_kd", double_t, 0, "x rate ctl PID D gain", 2, 0, 1000)

gen.add("y_kp", double_t, 0, "y ctl PID P gain", 0.5,  0, 100)
gen.add("y_ki", double_t, 0, "y ctl PID I gain", 0.1, 0,   100)
gen.add("y_kd", double_t, 0, "y ctl PID D gain", 0.0, 0, 100)
gen.add("vy_kp", double_t, 0, "y rate ctl PID P gain", 20,  0, 1000)
gen.add("vy_ki", double_t, 0, "y rate ctl PID I gain", 20, 0,  1000)
gen.add("vy_kd", double_t, 0, "y rate ctl PID D gain", 2, 0, 1000)

gen.add("z_kp", double_t, 0, "z ctl PID P gain", 1,  0, 100)
gen.add("z_ki", double_t, 0, "z ctl PID I gain",    0.001, 0,   100)
gen.add("z_kd", double_t, 0, "z ctl PID D gain",  0.1, 0, 100)
gen.add("vz_kp", double_t, 0, "z rate ctl PID P gain", 200, 0, 1000)
gen.add("vz_ki", double_t, 0, "z rate ctl PID I gain", 0, 0, 1000)
gen.add("vz_kd", double_t, 0, "z rate ctl PID D gain", 0, 0, 1000)

exit(gen.generate(PACKAGE, "mmuav_control", "UavPositionCtlParams"))
